File: uStock-Development-Android/app/src/main/java/data_structures/ai_models/ChatData.kt
package data_structures.ai_models

// This is a packaging file, packaging between views and sending to server.



import androidx.compose.runtime.mutableStateOf
import androidx.lifecycle.ViewModel

class ChatDataViewModel : ViewModel() {
    val botMessages = mutableStateOf<Map<String, List<BotMessage>>>(emptyMap())

    fun updateBotMessages(newBotMessages: Map<String, List<BotMessage>>) {
        botMessages.value = newBotMessages
    }
}

data class BotMessage(
    var message: String,
    var flag: Boolean
)

File: uStock-Development-Android/taskThree_models/Data Structures/AIModels/ChatData.swift
//
//  ChatData.swift
//  uStockIOS-V2-AI
//
//  Created by Gal Cohavy on 6/28/23.
//
// This is a packaging file, packaging between views and sending to server.

import Foundation



class ChatData: ObservableObject {
    var botMessages: [String: [BotMessage]] {
        didSet {
            objectWillChange.send()
        }
    }

    init(botMessages: [String: [BotMessage]] = [:]) {
        self.botMessages = botMessages
    }
}


struct BotMessage {
    var message: String
    var flag: Bool
}

